nifi-nar-bundles/nifi-hadoop-bundle/nifi-hdfs-processors/src/main/java/org/apache/nifi/processors/hadoop/PutHDFS.java

```diff
@@ -397,10 +432,28 @@ public void process(InputStream in) throws IOException {
                                     fos = codec.createOutputStream(fos);
                                 }
                                 createdFile = actualCopyFile;
-                                BufferedInputStream bis = new BufferedInputStream(in);
-                                StreamUtils.copy(bis, fos);
-                                bis = null;
-                                fos.flush();
+
+                                final String appendMode = context.getProperty(APPEND_MODE).getValue();
+                                if (APPEND_RESOLUTION.equals(conflictResponse)
+                                        && AVRO_APPEND_MODE.equals(appendMode)
+                                        && destinationExists) {
+                                    getLogger().info("Appending avro record to existing avro file");
+                                    try (final var reader = new DataFileStream<>(in, new GenericDatumReader<>());
```

Comment:
This won't backport to 1.x because `var` wasn't introduced until Java 10 and we still have to support Java 8 on 1.x for now.

Reply:
One additional note, use of `var` should be avoided in general, even when supported in the main branch.